### ISSUE TITLE
Error: Access Denied

### ISSUE THREAD

**Issue by Ale1x** [haydenbleasel/next-forge#298]
**Describe the bug**
Randomly I get "Error Access Denied" and can't use APP project

**next-forge version**
latest

**To Reproduce**
I get randomly this error. May occur sometimes when I refresh the page or trying go to another path

**Expected behavior**
Not getting the error

**Screenshots**
![image](https://github.com/user-attachments/assets/6377fd87-eda1-4da3-ac1d-e7c603b57710)
![image](https://github.com/user-attachments/assets/32525c28-1615-4f3a-b07a-4cda792f8fce)

**Desktop (please complete the following information):**
 - OS: Windows 10 PRO 22H2
 - Browser Brave v1.73.91

**Comment by Ale1x** [comment]
Fixed, was arcjet related, btw i suggest to add a check like this

`  if (process.env.NODE_ENV === 'production') {
    const req = await request();
    const decision = await aj.protect(req);

    if (decision.isDenied()) {
      if (decision.reason.isBot()) {
        throw new Error('No bots allowed');
      }
      throw new Error('Access denied');
    }
  }`
  
  Because in dev enviroment it can cause issues.

**Comment by davidmytton** [comment]
Hey @Ale1x - what error showed up in your Arcjet dashboard when making these requests? I'm curious to know which requests were denied and why so we can make sure this doesn't happen.

**Comment by Ale1x** [comment]
![image](https://github.com/user-attachments/assets/d6cca96f-8d3e-4433-9784-d7e7b0a2dd73)
![image](https://github.com/user-attachments/assets/1221fce7-7da9-4165-be92-c21dbb3a4474)

@davidmytton

**Comment by davidmytton** [comment]
Thanks @Ale1x. I tracked this down to an overly strict [Server Side Request Forgery](https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html) rule in the [Core Ruleset](https://github.com/coreruleset/coreruleset) which we use as part of our suspicious request analysis. This was applying rules in development which should only apply in production, so I improved our detection and added a test case.

The fix is rolling out now, so you shouldn't see this again. Thanks for helping debug it!
